client/my-sites/plans/jetpack-plans/product-store/items-list/all-items.tsx

```diff
@@ -91,18 +92,30 @@ export const AllItems: React.FC< AllItemsProps > = ( {
 						isSuperseded
 					);
 
+					// Go to the checkout page for all products when they click on the 'GET' CTA, except for Jetpack Social where we open a modal.
+					const ctaHref =
+						item.productSlug === PRODUCT_JETPACK_SOCIAL_BASIC &&
+						config.isEnabled( 'jetpack-social/advanced-plan' )
+							? '#'
```

Comment:
To make sure that the URL gets updated correctly and the page doesn't scroll to the top.

```suggestion
							? `#${ item.productSlug }`
```